rebasehelper/application.py

```diff
@@ -164,13 +164,15 @@ def _prepare_spec_objects(self):
         # check if argument passed as new source is a file or just a version
         if [True for ext in Archive.get_supported_archives() if self.conf.sources.endswith(ext)]:
             logger.verbose("argument passed as a new source is a file")
-            self.rebase_spec_file.set_version_using_archive(self.conf.sources)
+            main_source = self.spec_file._get_raw_source_string(  # pylint: disable=protected-access
```

Comment:
Since you are using the method in Application now, wouldn't it perhaps make sense to make it public?

Reply:
Maybe it would be better to make a wrapper method `SpecFile.get_main_source()`?
I didn't originally used the method at all, as I made `SpecFile.extract_version_from_archive_name()` non-static and with a single argument, but that proved problematic to unit-test.

Reply:
Creating `get_main_source` would be great to encapsulate the call, good suggestion